JAMath.py

```diff
@@ -40,15 +45,16 @@ def build_toolbar(self):
         separator2.props.draw = False
         separator2.set_expand(True)
         barra.insert(separator2, -1)
+        separator2.show()
 
         stop_button = StopButton(self)
-        stop_button.props.accelerator = '<Ctrl>q'
         barra.insert(stop_button, -1)
         stop_button.show()
+        stop_button.connect('clicked', self._stop_cb)
```

Comment:
I am just speculating, I am not sure how `Gtk` works exactly, but someone might explain

Connecting `stop_button` to `self._stop_cb` looks suspicious to me. I am not sure. `StopButton` internally sets the `connect` signal to `activity_close`, so I guess first `activity_close` would be executed before `_stop_cb` is called. Am I right?

Reply:
It doesn't matter what the callback order is.  `activity_close` does return.  What matters is `_stop_cb` is called, so that the event loop in the activity is terminated.  Otherwise what `activity_close` has done may take a while to happen.

Reply:
Yes, I agree; Resolving this conversation.

Reply:
What about using the method `can_close` , as it has been used in many other activities too?

Reply:
From the docstring documentation, `can_close` is used to check if the activity can be closed. It returns a boolean if the activity can be allowed to close. Using `activity_close` is not used to check if it can close but set the variable to False so that it breaks out from the game.

Reply:
What I would have done is, 
```python
def can_close(self):
    self._stop_cb()
    return True```

As the activity is synchronous, it would wait for self._stop_cb to be called, and then return True. Hmm, I am not sure which is the best implementation. Anyway, great its merged! :smile:

Reply:
I think that would go against the [description of and implied semantics of `can_close`](https://github.com/sugarlabs/sugar-toolkit-gtk3/blob/master/src/sugar3/activity/activity.py#L1174).